File: OPENAPI-REST-API/swagger-client/scala-akka-http-server/src/main/scala/io/swagger/server/model/CreateGeoFirewallRule.scala
package io.swagger.server.model


/**
 * = Create Firewall Rule =
 *
 * Create firewall rule for your ip
 *
 * @param destination_port  for example: ''22''
 * @param country_code To get country code refer our countries api - https://my.interserver.net/apiv2/account/countries?fetch_by=numcode for example: ''10''
 * @param asn ASN number for example: ''1331''
 * @param xdp_action 1 = Block,  0 = Whitelist for example: ''1''
 */
case class CreateGeoFirewallRule (
  destination_port: Option[Int],
  country_code: Option[Int],
  asn: Option[Int],
  xdp_action: Int
)

File: OPENAPI-REST-API/openapi-client/javascript-closure-angular/API/Client/CreateGeoFirewallRule.js
goog.provide('API.Client.CreateGeoFirewallRule');

/**
 * Create firewall rule for your ip
 * @record
 */
API.Client.CreateGeoFirewallRule = function() {}

/**
 * 1 = Block,  0 = Whitelist
 * @type {!number}
 * @export
 */
API.Client.CreateGeoFirewallRule.prototype.xdpAction;

/**
 * @type {!number}
 * @export
 */
API.Client.CreateGeoFirewallRule.prototype.destinationPort;

/**
 * To get country code refer our countries api - https://my.interserver.net/apiv2/account/countries?fetch_by=numcode
 * @type {!number}
 * @export
 */
API.Client.CreateGeoFirewallRule.prototype.countryCode;

/**
 * ASN number
 * @type {!number}
 * @export
 */
API.Client.CreateGeoFirewallRule.prototype.asn;

/** @enum {string} */
API.Client.CreateGeoFirewallRule.XdpActionEnum = { 
  0: '0',
  1: '1',
}
